src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -265,7 +275,7 @@ public void Abort()
         {
             EqtTrace.Verbose("TestRunRequest.Abort: Aborting.");
 
-            lock (this.syncObject)
+            lock (this.abortSyncObject)
```

Comment:
>abortSyncObject [](start = 23, length = 15)

is the usage of the sync object to synchronize the access to the executionmanager ?

Reply:
All execution related requests, like ExecuteAsync, HandleTestRunComplete, HandleTestRunStatsChange, HandleLogMessage are under same sync object.

But we want to respect cancel, abort and don't want to wait if some of the above request are going on. Thus separate lock objects for them

Reply:
Potentially reuse the same object for cancel and abort then?

Reply:
done.